app/assets/javascripts/miq_tree.js

```diff
@@ -98,11 +104,15 @@ function miqTreeActivateNode(tree, key) {
 
 // Activate silently (no onActivate event) and focus on a node within a tree given the node's key
 function miqTreeActivateNodeSilently(tree, key) {
-  var node = miqTreeFindNodeByKey(tree, key);
-  if (node) {
-    miqTreeObject(tree).selectNode(node, {silent: true });
-    miqTreeObject(tree).expandNode(node);
-    miqTreeScrollToNode(tree, key);
+  if (isReduxTree(tree)) {
+    ManageIQ.redux.store.dispatch({namespace: tree, type: 'node.check.silent', key: key});
```

Comment:
I'm not very happy about not using the constant here for dispatching the event. However, I don't see how else this could be done. @himdel might be able to tell us more.

Reply:
But a little semantics with the property order could help readability more:
1. type
2. namespace
3. key

Reply:
:thinking: :thinking: why is it check, when it's about selecting (highlighting) the node?

Reply:
The constants don't really help anything, I'm quite fine with just using the strings.

IMO we should be creating *methods* per action, not constants. That way, we actually get errors when trying to use an action which doesn't exist.

But that's not an existing convention, so, I'm fine with using a string.

That said .. please *please* keep in mind that the actions namespace is global.
"node.check.silent", is a bad name for an action if you don't know you're in the context of trees. IMO this should be `tree.node.check.silent` at least.

Reply:
Don't we want the `@@something/somethingelse` notation here instead?

Reply:
Yes, we resolved that in the chat, there will be @@tree/ something here, and will change the name, but for that, I need to actually modify the tree component itself first.